torch/testing/_internal/common_distributed.py

```diff
@@ -463,7 +470,7 @@ def init_multigpu_helper(world_size: int, backend: str):
     On a single node, all visible GPUs are evenly
     divided to subsets, each process only uses a subset.
     """
-    nGPUs = torch.cuda.device_count()
+    nGPUs = torch.xpu.device_count() if torch.xpu.is_available() else torch.cuda.device_count()
```

Comment:
This change introduced priority.
Can we determine which device count to use based on `backend: str` passed?
The parent function API:
```
def init_multigpu_helper(world_size: int, backend: str):
```

Reply:
Agreed. We will move those if..else in this PR  to make code generic.